### PR TITLE
Reduced complexity of cannon update function.

### PR DESCRIPTION
**Pull request by mauricewillemsen**
What is done:
stored the input for both players in an array which reduces half the code length. 
Stored the timer for left and right, the logging for the left and right in an array
made a method which chooses the step between left and right on input. So no left and right are in a loop

### CODE REVIEW

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java [hunk 1]
```diff
@@ -190,73 +206,30 @@ public final void update(final GameContainer container, final int delta,
...
-                }
-            } else {
-                this.timePassedKeyRight = 0;
-            }
-            if (container.getInput().isKeyDown(Input.KEY_LEFT)) {
-                if (container.getInput().isKeyPressed(Input.KEY_LEFT)) {
-                    Log.log(this, "Cannon moving to the left");
+        for (int i = 0; i < 2; i++) {
+            if (container.getInput().isKeyDown(INPUTKEY[playernr - 1][i])) {
+                this.timePassedKey[i] += delta;
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: Something I noticed. In 2 player mode: if one of the player presses the fire button, the timer is reset for both player. The both cannons will not auto shoot if at least pressed the fire button.

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java [hunk 2]
```diff
@@ -190,73 +206,30 @@ public final void update(final GameContainer container, final int delta,
...
-                if (this.timePassedKeyLeft > INPUT_SCAN_DELAY) {
-                    stepUp();
-                    this.timePassedKeyLeft = 0;
-                }
-            } else {
-                this.timePassedKeyLeft = 0;
             }
-            if (container.getInput().isKeyPressed(Input.KEY_W)) {
-                fire();
+            if (container.getInput().isKeyPressed(INPUTKEY[playernr - 1][i])) {
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: Can't you combine this If statement with If statement on line 210

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java [hunk 3]
```diff
@@ -181,6 +185,18 @@ public final void stepDown() {
     }
 
     /**
+     * Select the side of the cannon which should be updated.
+     * @param side integer of the side.
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: I would add a better description of the param. E.g. 0 = turn left. 1 = turn right

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java (hunk 2 above)
Comment (mauricewillemsen) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: No, one is for logging once when pressed, the other is for keep moving

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java (hunk 1 above)
Comment (mauricewillemsen) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: Tested it and i can't find the issue when playing, it works like it should.

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java (hunk 3 above)
Comment (mauricewillemsen) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: done

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java (hunk 2 above)
Comment (defnoob) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: The is KeyPressed is different from isKeyDown.

**Pull request merged**